deb-x64/Dockerfile

```diff
@@ -156,15 +157,10 @@ RUN curl -Sl -O https://dd-agent-omnibus.s3.amazonaws.com/kernel-4.9-headers-deb
     && rm kernel-4.9-headers-deb-x64.tgz
 
 # Go
-# Starting with go1.16, we need to build go from source, as the publicly distributed
-# go binaries are built with a linker (binutils) version that's higher than what we have in
-# the deb_x64 builder (the builder has 2.22, while go1.16 uses 2.26). Therefore, we need to
-# build go manually in the builder for it to work with the linker present in the builder image.
-# See: https://github.com/golang/go/issues/43996#issuecomment-782547917
-# TODO: investigate if we could use gvm to do this for us.
-COPY setup_go.sh /
```

Comment:
As mentionned on slack, I'd be in favor of removing the script since it isn't used anymore. If we ever need it we'll still need to revert this commit

Reply:
That's actually a good point, I realized that removing it in the same PR and the squashing would make it easier for people to find it back !